### ISSUE TITLE
[oxc_semantics] cfg does not handle ternary or chain expression control flow

### ISSUE THREAD

**Issue by TzviPM** [oxc-project/oxc#2289]
In general, we have chosen not to handle every expression explicitly in the CFG for performance reasons. That said, certain expressions also have implied control flow logic that we need to handle:

For example,  was `this` used before `super` in each of these cases?
- Ternary expression: `foo ? super() : null; this.a;`
    - if `foo` is false, then `super()` is not called
- Optional chain: `a?.(super()); this.a;`
    - if `a` is `undefined` or `null`, then `super()` is not called

**Comment by TzviPM** [comment]
I'll have a PR up shortly:

# Ternary (conditional) expression

Code:

```js
class A extends B {
  constructor() {
    const foo = bar ? super() : null;
    this.a();
  }
}
```

In `main` branch:

![main_ternary](https://github.com/oxc-project/oxc/assets/1950680/b8fb5c71-410f-4951-9d23-abfd38a3e780)

In my PR branch (pending):

![pr_ternary](https://github.com/oxc-project/oxc/assets/1950680/673c4b60-9a92-422e-bb16-0be2791650bc)

# Optional Chaining (chain) expression

Code:

```js
class A extends B {
  constructor() {
    foo?.(super());
    this.a();
  }
}
```

In `main` branch:

![main_chaining](https://github.com/oxc-project/oxc/assets/1950680/74d58a96-1620-4e55-82ec-b547637b215d)

In my PR branch (pending):

![pr_chaining](https://github.com/oxc-project/oxc/assets/1950680/a60835a7-b2d5-4a1d-912b-e4e5d4efe82a)

**Comment by rzvxa** [comment]
fixed via #3127
